### PR TITLE
024 registration system

### PR DESCRIPTION
**Pull request by MihneaBocaKainos**

### CODE REVIEW

File: app/controller/JobController.js [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
+    let s = await job.viewcompetency(bandID)
 
     res.render('competenciesperband', { comps: s, band: bandID })
 })
 
+router.get('/register', function(req, res){ 
+    res.render('register'); 
+});
+
+router.post('/register', async (req, res)=>{
```
Comment (hellofadude) [inline app/controller/JobController.js:26]: This route should be in its own file. As it is used for registering users perhaps `UserController` like back end would be appropriate

File: app/controller/JobController.js [hunk 2]
```diff
@@ -1,19 +1,38 @@
...
 
     res.render('competenciesperband', { comps: s, band: bandID })
 })
 
+router.get('/register', function(req, res){ 
+    res.render('register'); 
+});
+
+router.post('/register', async (req, res)=>{ 
+    var user = req.body
```
Comment (hellofadude) [inline app/controller/JobController.js:27]: Change to ECMA 6 - `const`

File: app/controller/JobController.js [hunk 3]
```diff
@@ -1,19 +1,38 @@
...
     res.render('competenciesperband', { comps: s, band: bandID })
 })
 
+router.get('/register', function(req, res){ 
+    res.render('register'); 
+});
+
+router.post('/register', async (req, res)=>{ 
+    var user = req.body 
+    encryptedPassword = await bcrypt.hash(user.password, 10);
```
Comment (hellofadude) [inline app/controller/JobController.js:28]: implicitly declare global variable. Can you explicitly declare it

File: app/service/UserService.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const axios = require('axios');
+const url = process.env.URL
+
+exports.registerUser = async (email, password, role) => { 
+    try {
+        const response = await axios.post(url+'/register', { "email": email, "password": password, "role": role })
```
Comment (hellofadude) [inline app/service/UserService.js:6]: Do these property names need to be in quotes?

File: app/views/register.html [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+      </div>
+    
+    </div>
+
+    <script>
+        var myInput = document.getElementById("password");
+        var letter = document.getElementById("letter");
+        var capital = document.getElementById("capital");
+        var special = document.getElementById("special");
+        var length = document.getElementById("length");
```
Comment (hellofadude) [inline app/views/register.html:83]: Use ECMA- 6 `let` or `const`

File: app/views/register.html [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    
+        <input class="govuk-input" type=email name="email" id="email" value="{{email}}" required/> <p></p>
+    
+        <strong>Password:</strong>
+    
+        <input class="govuk-input" type=password name="password" id="password" pattern="(?=.*[!@#$%^&*])(?=.*[a-z])(?=.*[A-Z]).{8,}" value="{{password}}" required><p></p>
+    
+
+    <label class="govuk-label" for="role" >Role:</label>
+    <select class="govuk-select" name="role" id="role" value="{{role}}">
```
Comment (hellofadude) [inline app/views/register.html:57]: I don't think value is a valid attribute for `select` html tag

File: app/views/register.html [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+        <p class="govuk-error-message" style="color: red;">Failed to register user!</p>
+      {% endif %}
+    
+    <h1 class="govuk-heading-xl">Register</h1>
+    
+    <form class="govuk-form-group" method="post">
+    
+        <strong>Email:</strong>
+    
+        <input class="govuk-input" type=email name="email" id="email" value="{{email}}" required/> <p></p>
```
Comment (hellofadude) [inline app/views/register.html:49]: Line 49 and 53. You can associate a new label with input element for accessibility reasons for instance 

 `<label for="email"></label><input class="govuk-input" type=email name="email" id="email" value="{{email}}" required/> <p></p>`

File: test/Unit/Service/UserServiceTest.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+const user ={
+    email:"Something@anything.com",
+    password:"Password!",
+    role:"Admin"
+}
+
+describe('UserService', function () {
+    describe('registerUser', function () {
+        it('should return error 500', async () => {
+            var mock = new MockAdapter(axios);
```
Comment (hellofadude) [inline test/Unit/Service/UserServiceTest.js:17]: Use ECMA Script 6 `let` or `const`

**Review by hellofadude — APPROVED** [review]

**Review by Ryan-Jones1999 — APPROVED** [review]
Looks good cloned down and was able to register and see it in db encrypted etc

**Pull request merged**